straxen/plugins/veto_events.py

```diff
@@ -473,3 +461,99 @@ def get_window_size(self):
 
     def compute(self, hitlets_mv, start, end):
         return super().compute(hitlets_mv, start, end)
+
+
+@strax.takes_config(
+    strax.Option('hardware_delay_nv', default=0, type=int,
+                 help="Hardware delay to be added to the set electronics offset."),
+)
+class nVETOEventsSync(strax.Plugin):
+    """
+    Plugin which computes time stamps which are synchronized with the
+    TPC. Uses delay set in the DAQ.
+    """
+    depends_on = 'events_nv'
+    provides = 'events_sync_nv'
+    save_when = strax.SaveWhen.EXPLICIT
+    __version__ = '0.0.1'
+
+    def infer_dtype(self):
+        dtype = []
+        dtype += strax.time_fields
+        dtype += [(('Time of the event synchronized according to the total digitizer delay.',
+                    'time_sync'), np.int64),
+                  (('Endtime of the event synchronized according to the total digitizer delay.',
+                    'endtime_sync'), np.int64),
+                  ]
+        return dtype
+
+    def setup(self):
+        self.total_delay = get_delay(self.run_id)
+        self.total_delay += self.config['hardware_delay_nv']
+
+    def compute(self, events_nv):
+        events_sync_nv = np.zeros(len(events_nv), self.dtype)
+        events_sync_nv['time'] = events_nv['time']
+        events_sync_nv['endtime'] = events_nv['endtime']
+        events_sync_nv['time_sync'] = events_nv['time'] + self.total_delay
+        events_sync_nv['endtime_sync'] = events_nv['endtime'] + self.total_delay
+        return events_sync_nv
+
+
+def get_delay(run_id):
+    """
+    Function which returns the total delay between TPC and veto for a
+    given run_id. Returns nan if
+    """
+    try:
+        import utilix
+    except ModuleNotFoundError:
+        return np.nan
+
+    delay = np.nan
+    if straxen.utilix_is_configured():
+        run_db = utilix.DB()
+        run_meta = run_db.get_doc(run_id)
```

Comment:
is there a strong reason why you are the API instead of mongo itself. It sounds unnecessarily slow?
https://github.com/XENONnT/utilix/blob/master/utilix/rundb.py#L177